website/blog/2023-05-28-databend-weekly-95.md

```diff
@@ -0,0 +1,160 @@
+
+Here are some code snippets that may help you understand how to use PyO3.
+
+In `asyncio.rs`, a Struct called `AsyncDatabendDriver` can be defined as a Python class, and the Python module can be defined as `databend-driver`. 
+
+```rust
+/// `AsyncDatabendDriver` is the entry for all public async API
+#[pyclass(module = "databend_driver")]
+pub struct AsyncDatabendDriver(Connector);
+```
+
+The corresponding methods for `AsyncDatabendDriver` are implemented next, with underlying calls made to the methods implemented in Rust's Trait. For example, in this code snippet, we see that `exec` is used:
+
+```rust
+#[pymethods]
+impl AsyncDatabendDriver {
+    #[new]
+    #[pyo3(signature = (dsn))]
+    pub fn new(dsn: &str) -> PyResult<Self> {
+        Ok(AsyncDatabendDriver(build_connector(dsn)?))
+    }
+
+    /// exec
+    pub fn exec<'p>(&'p self, py: Python<'p>, sql: String) -> PyResult<&'p PyAny> {
+        let this = self.0.clone();
+        future_into_py(py, async move {
+            let res = this.connector.exec(&sql).await.unwrap();
+            Ok(res)
+        })
+    }
+}
+```
+
+Finally, add `AsyncDatabendDriver` as a Python class in `lib.rs`:
+
+```rust
+#[pymodule]
+fn _databend_driver(_py: Python, m: &PyModule) -> PyResult<()> {
+    m.add_class::<AsyncDatabendDriver>()?;
+    Ok(())
+}
+```
+
+If you are interested in learning more, please check out the resources listed below:
+
+- [PR | feat: Support python binding](https://github.com/datafuselabs/bendsql/pull/112)
+
+## Highlights
+
+Here are some noteworthy items recorded here, perhaps you can find something that interests you.
+
+- Added table lock for compact.
+- Added `system.indexes` and `show indexes` command.
+- Added support for azure openai api.
+- Added new bitmap-related functions: `bitmap_and_count`, `bitmap_or_count`, `bitmap_xor_count`, `sub_bitmap`, `bitmap_subset_limit` and `bitmap_subset_in_range`.
+- Added new window functions: `lag`, `lead`, `first_value`, `last_value` and `nth_value`.
+
+## What's Up Next
+
+We're always open to cutting-edge technologies and innovative ideas. You're more than welcome to join the community and bring them to Databend.
+
+### Support Custom Character Sets
+
+A character set is a set of rules that defines how characters are represented, while collation determines the order of characters within the character set. Character sets are used to store data and can contain multiple collations.
+
+It would be great to have support for custom character sets in the same way as MySQL. e.g. IBM z/OS uses EBCDIC encoding by default.
```

Comment:
It would be great to have support for custom character sets in the same way as MySQL, e.g., IBM z/OS uses EBCDIC encoding by default.